instrumentation/armeria-0.99/library/src/main/java/io/opentelemetry/instrumentation/armeria/v0_99/server/ArmeriaServerTracer.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.armeria.v0_99.server;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.util.Version;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import io.grpc.Context;
+import io.opentelemetry.auto.bootstrap.instrumentation.decorator.HttpServerTracer;
+import io.opentelemetry.context.propagation.HttpTextFormat.Getter;
+import io.opentelemetry.instrumentation.armeria.v0_99.internal.ContextUtil;
+import io.opentelemetry.trace.Tracer;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.URI;
+
+final class ArmeriaServerTracer
+    extends HttpServerTracer<HttpRequest, ServiceRequestContext, RequestContext> {
```

Comment:
it looks like we aren't using the `STORAGE` parameter, e.g. no calls to `getServerContext`

if that's the case, maybe good to be explicit about this (similar to `AkkaHttpServerTracer`)

```suggestion
    extends HttpServerTracer<HttpRequest, ServiceRequestContext, Void> {
```

Reply:
ya, it's attached in `startScope`, but then never read after that

Reply:
Hmm I think I want to keep it like this, Armeria does provide a storage implementation, whether that's actually used or not is more a decision of `HttpServerTracer` than the armeria instrumentation from what I can tell.

Reply:
> whether that's actually used or not is more a decision of `HttpServerTracer` than the armeria instrumentation from what I can tell

so, confusingly, this is not the case. it's up to the individual instrumentation to use what's in `STORAGE` if they need it for propagation

it seems likely we will need it in the armeria instrumentation in the future, e.g. for propagating context to some async handlers, so we can keep it, though maybe a comment about it not being used currently?

@iNikem please correct me if I'm missing something about the `STORAGE` param (either current or future plans).

Reply:
oh yes, and i opened #791 to document the `STORAGE` param, it is confusing when/if to use it

Reply:
Ok less complexity makes sense. I'll revisit after instrumenting client and introducing some more complex async scenarios.